Rocknix RK3566-X55 mainline

ananjaser1211 opened the pull request:
## Description

Note : This is an initial PR to bring the RK3566-X55 Mainline back into Rocknix

- This includes changes and patches needed to compile and boot RK3566-X55 Device on latest rocknix dev branch
- Uses a "new" Singleadc joystick driver with support for X55's 4 SARADC channels
- The controller "retrogame_joypad" uses the same GUID as other RK3566 Devices and does not require custom controller mappings, so all of it is symlinked 

## Current Issues on RK3566-X55
- Bluetooth Controller support (DS4 connects but thats about it, Audio works through CLI pairing)

## Type of change

Please delete options that are not relevant.

- [X] Device support

## How Has This Been Tested Locally?

- [X] Test A
- [ ] Test B

- Successful Compilation
- boot into emulation station
- run games through RetroArch and PPSSPP
- Test Volume/Volume Change
- Test deep Sleep functionality
- Test WiFi connectivity & SSH
- Test Brightness change
- Test Device LED options
- Test Controller in various emulators (Mupen, Drastic, PPSSPP, Duckstation, Dreamcast, Retroarch)
- Test installation to EMMC (+ @sydarn installer patch)
- Test Portmaster (2024.04.01-1021)
- Test Bluetooth Audio

**Test Configuration**:
* Build OS name and version: Linux Mint 21 (22.04)
* Docker (Y/N): N
* ROCKNIX Branch: dev

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

r3claimer commented:
Looks good. The only thing I see is that some of these config files could probably just be symbolic links instead of their own dir. like the RA ones.

sydarn commented:
Unless there is some inherent issue with it, i would prefer to have linux ext conf as a symbolic link as well. I think it should be possible to have them compatible.

ananjaser1211 commented:
At the moment the only differences in ext linux is the joystickdriver

X55s driver is `CONFIG_JOYSTICK_SINGLEADC_QUADSAR_JOY` other than that all configs are identical,

Regarding RA, its only for the re-scale [factor ](https://github.com/ROCKNIX/distribution/pull/63/commits/370e047efa1391fc8fca59b93283a9fe174cd439) )so the display is not zoomed in, i have not tried the most recent 3566 configs so will see how it goes, if the issue is resolved i will symlink RA

The U-BOOT patches should be symlinked to RK3566, as it already has the pkx55 patch and so on

ananjaser1211 commented:
Rebased and got rid of device-specifc RA and u-boot patches as they are indeed not required, i modified RK3566 `menu_widget_scale_factor` to avoid over sized RA Achievements popups etc, should be tested on other RK3566 devices though.

will keep it a draft for now until i figure out why it stopped booting on latest rebase

ananjaser1211 commented:
As mentioned in https://github.com/ROCKNIX/distribution/pull/63/commits/22fd74703ec5ba3d5e2920470ad796570fa4552a , emmc installation can cause issues that are hard to recover from, at the moment once system is installed on emmc it will no longer be bootable from sdcard, i dont know why yet, the safer option is to disable it until that is sorted.

otherwise, the PR is ready.

sydarn commented:
did you try to find a way to derive or symlink the ext linux config?

sydarn commented:
THANKS! :)
